TESTS/mbed_hal/sleep_manager/main.cpp

```diff
@@ -189,7 +224,7 @@ void test_lock_unlock_test_check()
     // Deep sleep unlocked:
     // * sleep_manager_can_deep_sleep() returns true,
     // * sleep_manager_can_deep_sleep_test_check() returns true instantly.
-    TEST_ASSERT_TRUE(sleep_manager_can_deep_sleep());
+    TEST_ASSERT_TRUE(sleep_manager_can_deep_sleep_test_check());
```

Comment:
I assume this is to cover CMPOK interrupt, correct? Instead of this, could you add a `wraparound_lp_protect()` call above, please?

This test case confirms that `sleep_manager_can_deep_sleep_test_check()` can actually replace the original `sleep_manager_can_deep_sleep()` in some greentea tests. The *_test_check* has to be compared with the original.

https://github.com/ARMmbed/mbed-os/blob/9974899a6b1947f0ad4c729a3608b2d172d6eb28/platform/mbed_power_mgmt.h#L150-L159

Reply:
wraparound_lp_protect() will not ensure that CMPOK has fired. It will just check that the counter is not to close to the wrap around limit (e.g. 0xFFFF). An active wait could maybe be used instead ?

Reply:
@fkjagodzinski good point. I will look into this tomorrow.